lnd.go

```diff
@@ -213,15 +213,42 @@ func Main(cfg *Config, lisCfg ListenerCfg, implCfg *ImplementationCfg,
 		return mkErr("error initializing DBs: %v", err)
 	}
 
+	var serverOpts []grpc.ServerOption
+	var restDialOpts []grpc.DialOption
+	var restListen func(net.Addr) (net.Listener, error)
+	var cleanUp func()
+
 	tlsManager := NewTLSManager(cfg)
-	// Only process macaroons if --no-macaroons isn't set.
-	serverOpts, restDialOpts, restListen, cleanUp,
-		err := tlsManager.getConfig()
+	// If the TLSEncryptKey setting is on, we need to generate an ephemeral
+	// certificate here until we're able to use until the wallet is
+	// unlocked and a new TLS pair can be encrypted to disk. Otherwise we
+	// can process the certificate normally.
+	if cfg.TLSEncryptKey {
+		_, err := tlsManager.LoadEphemeralCertificate()
+		if err != nil {
+			return mkErr("unable to load ephemeral certificate",
+				err)
+		}
 
-	if err != nil {
-		return mkErr("unable to load TLS credentials: %v", err)
+		serverOpts, restDialOpts, restListen,
+			err = tlsManager.getConfig()
+		if err != nil {
+			return mkErr("unable to load TLS credentials: %v", err)
+		}
+	} else {
+		_, cleanUp, err := tlsManager.generateOrRenewCert()
```

Comment:
The `getConfig` and `generateOrRenewCert` methods should be exported so we could eventually even move the TLS manager to a different package.